flows/actions/call_classifier.go

```diff
@@ -0,0 +1,130 @@
+package actions
+
+import (
+	"encoding/json"
+
+	"github.com/nyaruka/goflow/assets"
+	"github.com/nyaruka/goflow/flows"
+	"github.com/nyaruka/goflow/flows/events"
+	"github.com/pkg/errors"
+)
+
+func init() {
+	registerType(TypeCallClassifier, func() flows.Action { return &CallClassifierAction{} })
+}
+
+const (
+	classificationCategorySuccess = "success"
+	classificationCategorySkipped = "skipped"
+	classificationCategoryFailure = "failure"
+)
+
+var classificationCategories = []string{classificationCategorySuccess, classificationCategorySkipped, classificationCategoryFailure}
+
+// TypeCallClassifier is the type for the call classifier action
+const TypeCallClassifier string = "call_classifier"
+
+// CallClassifierAction can be used to classify the intent and entities from a given input using an NLU classifier. It always
+// saves a result indicating whether the classification was successful, skipped or failed, and what the extracted intents
+// and entities were.
+//
+//   {
+//     "uuid": "8eebd020-1af5-431c-b943-aa670fc74da9",
+//     "type": "call_classifier",
+//     "classifier": {
+//       "uuid": "1c06c884-39dd-4ce4-ad9f-9a01cbe6c000",
+//       "name": "Booking"
+//     },
+//     "input": "@input.text",
+//     "result_name": "Intent"
+//   }
+//
+// @action call_classifier
```

Comment:
@nicpottier @ericnewcomer 🚨 name tweak again...

I had this called `classify_text` and the event being `text_classified`.. but this feels more accurate. This action by itself doesn't really classify text... it pulls data in from an external source. The entire node (this action plus a router) is more accurately the thing that does classification. It also generates an event that records an HTTP event rather than a classification.. so trying out a naming scheme that puts it with the webhook stuff...

`call_webook` -> `webhook_called`
`call_resthook` -> `webhook_called` x N
`call_classifier` -> `classifier_called`

Reply:
That does feel consistent, one thing to think about is what happens if we plug in speech to text. Wit treats this as just another intent call of sorts..

IE: https://wit.ai/docs/http/20160526#post__speech_link

Maybe that's a totally different thing though, Luis doesn't seem to tie these as closely together. So I'm cool with `call_classifier`

Reply:
Tho was about to get rid of the `Skipped` result category and now it feels a little dishonest for and action called `call_classifier` to create a `Success` result if it didn't really call the classifier... do you think you could learn to love `Success|Skipped|Failure` 😬 ?

Reply:
I'm fine with it, just always feel that when we get into the situation of looking at more than one status invariably we add another status and then that breaks things. But agree that may not happen. I just don't love it, but that's ok.

Reply:
To be clear, this skipped scenario is if the input is empty or there is a misconfiguration and we want flow authors to route that possibility on every classify node?

Reply:
So we're saying there's a contract for these calley actions that they always save a result.. because then that makes it cleaner for routing. Don't have to handle missing results. But there are going to be cases where the call action doesn't actually call anything (e.g. no input to classify, or an airtime transfer of zero).. in which case it saves a result to allow routing to proceed... but it's not quite call=success or call=failure

Reply:
Got it. Just doing my job to over-communicate that we are asking users to _always_ route what very well might _always_ be an edge case for them. I don't disagree with the reasoning.